supervision/detection/vlm.py

```diff
@@ -406,3 +410,78 @@ def from_google_gemini(
         return np.empty((0, 4)), np.empty((0,), dtype=str)
 
     return np.array(xyxy), np.array(labels)
+
+
+def from_moondream(
+    result: dict,
+    resolution_wh: Tuple[int, int],
+) -> Tuple[np.ndarray, np.ndarray]:
+    """
+    Parse and scale bounding boxes from moondream JSON output.
+
+    The JSON is expected to have a key "objects" with a list of dictionaries:
+      {
+          "objects": [
+              {"x_min": 0.1, "y_min": 0.2, "x_max": 0.3, "y_max": 0.4},
+              ...
+          ]
+      }
+
+      For Example:
+      {
+          "objects": [
+              {"x_min": 0.1, "y_min": 0.2, "x_max": 0.3, "y_max": 0.4},
+              {"x_min": 0.5, "y_min": 0.6, "x_max": 0.7, "y_max": 0.8}
+          ]
+      }
+
+
+    Args:
+        result: Dictionary containing the JSON output from the model.
+        resolution_wh: (output_width, output_height) to which we rescale the boxes.
+
+    Returns:
+        xyxy (np.ndarray): An array of shape `(n, 4)` containing
+            the bounding boxes coordinates in format `[x1, y1, x2, y2]`
+        class_id (Optional[np.ndarray]): An array of shape `(n,)` containing
+            the class indices for each bounding box (or None if `classes` is not
+            provided)
+        class_name (np.ndarray): An array of shape `(n,)` containing
+            the class labels for each bounding box
+    """  # docs
+
+    w, h = resolution_wh
+    if w <= 0 or h <= 0:
+        raise ValueError(
+            f"Both dimensions in resolution_wh must be positive. Got ({w}, {h})."
+        )
+
+    if "objects" not in result or not isinstance(result["objects"], list):
+        return np.empty((0, 4)), np.empty((0,), dtype=int)
+
+    denormalize_xyxy = []
+
+    for item in result["objects"]:
+        if not all(k in item for k in ["x_min", "y_min", "x_max", "y_max"]):
+            continue
+
+        x_min = item["x_min"]
+        y_min = item["y_min"]
+        x_max = item["x_max"]
+        y_max = item["y_max"]
+
+        denormalize_xyxy.append(
+            denormalize_boxes(
+                np.array([x_min, y_min, x_max, y_max]).astype(np.float64),
+                resolution_wh=(w, h),
+            )
+        )
+
+    if not denormalize_xyxy:
+        return np.empty((0, 4)), np.empty((0,), dtype=int)
+
+    xyxy = np.array(denormalize_xyxy, dtype=float)
+    num_detections = len(xyxy)
+    class_id = np.arange(num_detections)
```

Comment:
the class_id is the index?

Reply:
I used index to save people use "color lookup" in annotators. But there is no id or class_name

Reply:
Okay I removing it.